### DISCUSSION TITLE
Is/will Cruiser (be) prepared for Carplay or Android Auto?

### DISCUSSION THREAD

**Discussion by DIrgi** [devemux86/cruiser#508] [Q&A] [answered]
(example pictures from carpuride page)
<img width="464" alt="image" src="https://github.com/devemux86/cruiser/assets/57968526/09dafa14-7587-475c-8f83-2a67f81d9821">
![image](https://github.com/devemux86/cruiser/assets/57968526/ce41c97e-9073-4dca-b74c-d4b5fa33c58a)
![image](https://github.com/devemux86/cruiser/assets/57968526/db7b89f7-4de6-4c9e-bf91-1882d54bec18)
![image](https://github.com/devemux86/cruiser/assets/57968526/668e4e95-cb39-435b-a284-c86e7849c9f6)

Hi, 
the last few weeks, I see more and more discussions and topics in forums and YT videos  around **CarPlay** since at least a asian company, [carpuride](https://carpuride.com/) (and maybe others) offers relatively cheap outdoor devices to have CarPlay or Android Auto on your motorbike.
The advantage is clear, the cell phone is streaming his screen and functions onto this remote display, need not to be placed on the bike and can remain and perfect sheltered in pocket - I assume, this will be somehow the future like you have it already in modern cars. (btw. the display works already on most modern BMW navigation craddles (with an adapter), where you normally place your high expensive ;-) Garmin build BMW Navigator device)

Never the less the disadvantage is, that not only a smartphone has to support this kind of connection where "CarPlay" is iPhone driven and "Android Auto" well, Android driven - **The software or App itself has to support this** as well and there I see the main issue or barrier since I assume all modern smartphone from hardware side already support these connections.
If it's not supported by the app, it cannot be used with the app or the functions are limited.

**Question to Emux:**
1. Does Cruiser work already with "Android Auto"? (CarPlay is iOS based apps)
2. If not, do you think about to make it possible (midterm)?

As mentioned above, I see a growing interest for future solutions, as well to make Cruiser more popular - one robust (cheap) outdoor display for everything - When changing the smartphone you need noting to change on your bike, just connect and go ahead. You need not to have a focus on a robust smartphone or tablet since it can be stored somewhere in a safe, dry pocket. You need not to have any additional bracket somewhere where you have to have a power connection as well. Most easy way then is charging via power bank somewhere in pocket.

Cheers, Dirk

**Comment by devemux86** [comment, marked as answer]
@DIrgi thanks for the suggestion.

Android Auto usually requires writing a new app or rewriting large parts,
because it has different requirements and too many restrictions.

App features and user experience can never be as rich as in regular Android.

Cruiser is only written by me (1 developer) and this needs a lot of work.
It's in my list, however it's difficult to make any promises right now. :slightly_smiling_face:

**Reply by DIrgi** [reply]
I understand, I just want to flag this topic and asked for your assessment and estimation.
I fully understand that's it's not easy for a one man show and I don't know the effort behind to make this possible.
But...since it's in your list.....maybe on one day...anyways it just started to become slightly more popular.
Let's see what the future bring. I bought my outdoor tablet half away ago and I didn't had CarPlay or Android Auto in my mind at this time.

**Comment by devemux86** [comment]
I'm also thinking of making Cruiser open source.

If there is interest from the community,
more users can contribute with new features and improvements.

**Reply by MotoUKRider** [reply]
Interesting!  There's a chance I may give Android dev a try - there's not actually anything (other than a nav app) that I've really wanted to try.  I did give the "app inventor" a go some years back but that's not *really* coding :smile:  I made a simple GPS "ping" on a timer that pushed my location to a PHP file on my webserver - family could then load a private webpage and see where I was "on tour".

My experience goes back to BASIC, MS-DOS and VB/NET and C++ with Arduino.  If the past month of learning mkgmap/osmium and the joys of TYP files has taught me anything, it's that if you're determined, you can succeed!  My learning seems to be based on "look at an example app/code and then work out what each bit does" so I like to use a sample project and pick it apart rather than starting with a "hello world" app.

If you just wanted help with certain aspects, and I had a decent understanding then I'd volunteer but I think all I could do at present is make a cosmetic change (I prefer GUI work to coding) and re-compile rather than add any fancy new features as quick as you do!

**Reply by devemux86** [reply]
I created a new topic about the open source possibility, please let's discuss it there:
- https://github.com/devemux86/cruiser/discussions/509

**Reply by DIrgi** [reply]
@devemux86 Interested  yes but....
to be honest more then the contribution I do right now and I don't know if and how deep I'll continue, I cannot dive into Android or linux programming. I tried it once with a previous project from a fellow here in the net who used the OSM raw data to create a complete motorbike like navigation map for Garmin with a linux backend.
He asked me to port this to windows, so I started wit VBstudio - I'm somehow a little familiar with VB or VBA (via Excel), but compare to former times. World get's more and more complicated even VBAstudio became a colossus where it was not easy to keep an overview.

The project itself, "Motoroute" was great and the outcome as well, I used it for years on my Garmin (via Linux. subsystem on Windows, you could even use "Windows" to generate your own desired map. 
It was unbeatable especially when it came to offroad navigation where Garmin simply quits or you had to by expensive Topo mp which finally didn't cover every are where you maybe like to cruise around.

He asked me as well If I could contribute. But it was finally a nightmare. I spend hours/days to dive into a new programming language and my wife started to complain that I spent night/and day on the desktop. So...after I did some longer breaks and I had to start most of the time again from scratch since my less "new knowledge" had already been gone. So I had to skip my participation unfortunately.

Since the former developer and programmer for Motoroute didn't find a follow up, he closed the project 2,3 years ago....
It was sad but what can you do, when you are getting retired and have no successor.
